pkg/controller/operators/olm/operator.go

```diff
@@ -1627,7 +1627,7 @@ func (a *Operator) operatorGroupForCSV(csv *v1alpha1.ClusterServiceVersion, logg
 			return nil, nil
 		}
 		logger.Debug("csv in operatorgroup")
-		return operatorGroup, nil
```

Comment:
Any idea on whether we should be handling this kind of operation here vs. any call sites to this function? If I'm reading the code right, we get a list of OperatorGroup resources in the CSV's namespace, and all of those OperatorGroup resources should be treated as read-only resources, so it might be reasonable to return a copy of that resource if the blast radius is high.